### ISSUE TITLE
CIO server always start on "0.0.0.0" - does not respect "connector" configuration

### ISSUE THREAD

**Issue by dkrivoruchko** [ktorio/ktor#1959]
**CIO Server ktor-server-cio:1.3.2-1.4-M2**
 
**Describe the bug**
CIO server always start on "0.0.0.0" - does not respect "connector" configuration

**To Reproduce**
```
embeddedServer(CIO, applicationEngineEnvironment {
            module { ... }
            connector {
                host = "192.168.0.1"
                port = 8080
            }
})
```
Server will start on "0.0.0.0:8080" instead of "192.168.0.1:8080"

**Expected behavior**
Server respect connector configuration.

**Comment by dkrivoruchko** [comment]
CIOApplicationEngine.startConnector - must configure itself based on connector host and port. Currently only port is set.

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-334">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Fixed in 1.5.2
